### PR TITLE
Ruby on Rails group project

### PR DESCRIPTION
**Pull request by TaiwoEnoch**
### 
<h1>**Hello Code Reviewer 👋 👋**</h1>

## I @TaiwoEnoch and my coding partners @karanJ2212 and @Rishi-Mishra0704 create this PR to feature all requirements for the Recipe app as a RoR group project.
# Project requirements
- [ ] You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
- [ ]  Login page and registration page:
- [ ]  Should be built with Devise.
- [ ]  Food list (only for a group that has less than 3 members):
- [ ]  Should display a list of food added by the logged-in user as in the wireframe (display also quantity of a given food).
- [ ]  Should lead to a form that allows users to add new food.
- [ ]  Recipes list:
- [ ]  Should display a list of recipes created by the logged-in user as in the wireframe.
- [ ]  Should lead to recipe details.
- [ ]  If the user is the owner of the recipe, should allow the user to delete it.
- [ ]  Public recipe list:
- [ ]  Should display a list of all public recipes ordered by newest as in the wireframe.
- [ ]  Should lead to recipe details.
- [ ]  Recipe details:
- [ ]  Should display a toggle button that allows for a recipe to be made public or private.
- [ ]  If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
- [ ]  If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- [ ]  General shopping list view (only for a group that has less than 3 members):
- [ ]  Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
- [ ]  Should count the total food items and total price of the missing food.
- [ ]  Make sure there are no N+1 queries happening.
- [ ]  Create a navigation menu that allows users to open all of the pages you created.
- [ ]  Write unit and integration tests
- [ ]  If your team has 3 members these are requirements; otherwise, they are additional requirements
- [ ]  Inventories list:
- [ ]  Should display a list of inventories created by the logged-in user as in the wireframe.
- [ ]  Should lead to inventory details.
- [ ]  If the user is the owner of the inventory, should allow to delete it.
- [ ]  Inventory details:
- [ ]  Should display the inventory details as in the wireframe.
- [ ]  Should lead to a form that allows users to add new food.
- [ ]  Recipe details:
- [ ]  Should have a modal to choose an inventory to create a shopping list with, as in the wireframe.
- [ ]  Inventory shopping list:
- [ ]  Should show the list of food that is missing by comparing the food in the recipe with the food in the inventory.
- [ ]  Should count the total food items and total price of the missing food.

### CODE REVIEW

File: app/views/recipes/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    <% @recipes.each do |recipe| %>
+      <li class="card row-wrap public-recipe">  
+        <div class="head">
+          <a href=<%= "recipes/#{recipe.id}" %> class="no-styling">
+            <h3><%= recipe.name %></h3>
+          </a>
+           <a href=<%= "recipes/#{recipe.id}" %> class="no-styling text-description">
+          <p class="text-description"><%= recipe.description %></p>
+        </a>
+        </div>
```
Comment (Owusu-Desmond) [inline app/views/recipes/index.html.erb:21]: - [ ] Please make sure that your recipes list page layout design matches the provided wireframe below :point_down: 
  
  ### Your design
  image
  
  ### Provide wireframe
  image

File: app/models/recipe.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class Recipe < ApplicationRecord
+  belongs_to :user
+  has_many :recipe_foods
+  has_many :foods, through: :recipe_foods
+
+  scope :public_recipes, -> { where(public: true) }
+
```
Comment (Owusu-Desmond) [inline app/models/recipe.rb:7]: - [ ] Please make sure that when a recipe is switched from public to private, it is saved in the database and it should not show on the public recipes page :pray: 
  
  [Screencast from 07-15-2023 08:30:25 PM.webm](https://github.com/TaiwoEnoch/FoodyOnRails/assets/92303891/fe75bb7c-d587-4970-b8ca-815e66d4fcff)

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @TaiwoEnoch, @Rishi-Mishra0704 and @karanJ2212 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Rishi-Mishra0704** [comment]
@Owusu-Desmond we have applied the changes please review it

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @TaiwoEnoch && @Rishi-Mishra0704 && @karanJ2212  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Changes made as requested by the previous reviewer :+1: 
- The app is working as expected :surfer: 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**